public/app/features/serviceaccounts/ServiceAccountPage.tsx

```diff
@@ -64,79 +56,196 @@ const ServiceAccountPageUnconnected = ({
-  deleteServiceAccountToken,
-  deleteServiceAccount,
-  updateServiceAccount,
-  fetchACOptions,
-}: Props) => {
-  const [isModalOpen, setIsModalOpen] = useState(false);
+}: Props): JSX.Element => {
+  const dispatch = useDispatch();
   const [newToken, setNewToken] = useState('');
+  const [isTokenModalOpen, setIsTokenModalOpen] = useState(false);
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
+  const [isDisableModalOpen, setIsDisableModalOpen] = useState(false);
+  const styles = useStyles2(getStyles);
+  const serviceAccountId = parseInt(match.params.id, 10);
+  const tokenActionsDisabled =
+    !contextSrv.hasPermission(AccessControlAction.ServiceAccountsWrite) || serviceAccount.isDisabled;
+
+  const ableToWrite = contextSrv.hasPermission(AccessControlAction.ServiceAccountsWrite);
 
   useEffect(() => {
-    const serviceAccountId = parseInt(match.params.id, 10);
-    loadServiceAccount(serviceAccountId);
-    loadServiceAccountTokens(serviceAccountId);
+    dispatch(loadServiceAccount(serviceAccountId));
+    dispatch(loadServiceAccountTokens(serviceAccountId));
     if (contextSrv.licensedAccessControlEnabled()) {
-      fetchACOptions();
+      dispatch(fetchACOptions());
     }
-  }, [match, loadServiceAccount, loadServiceAccountTokens, fetchACOptions]);
+  }, [match, dispatch, serviceAccountId]);
+
+  const showDeleteServiceAccountModal = (show: boolean) => () => {
+    setIsDeleteModalOpen(show);
+  };
+
+  const showDisableServiceAccountModal = (show: boolean) => () => {
+    setIsDisableModalOpen(show);
+  };
+
+  const handleServiceAccountDelete = () => {
+    dispatch(deleteServiceAccount(serviceAccount.id));
+  };
+
+  const handleServiceAccountDisable = () => {
+    dispatch(updateServiceAccount({ ...serviceAccount, isDisabled: true }));
+    setIsDisableModalOpen(false);
+  };
+
+  const handleServiceAccountEnable = () => {
+    dispatch(updateServiceAccount({ ...serviceAccount, isDisabled: false }));
+  };
 
   const onDeleteServiceAccountToken = (key: ApiKey) => {
-    deleteServiceAccountToken(parseInt(match.params.id, 10), key.id!);
+    dispatch(deleteServiceAccountToken(serviceAccount?.id, key.id!));
   };
 
   const onCreateToken = (token: ServiceAccountToken) => {
-    createServiceAccountToken(serviceAccount.id, token, setNewToken);
+    dispatch(createServiceAccountToken(serviceAccount?.id, token, setNewToken));
   };
 
-  const onModalClose = () => {
-    setIsModalOpen(false);
+  const onTokenModalClose = () => {
+    setIsTokenModalOpen(false);
     setNewToken('');
   };
 
   return (
     <Page navModel={navModel}>
       <Page.Contents isLoading={isLoading}>
         {serviceAccount && (
-          <>
-            <ServiceAccountProfile
-              serviceAccount={serviceAccount}
+          <div className={styles.headerContainer}>
+            <a href="org/serviceaccounts">
+              <IconButton size="xxl" variant="secondary" name="arrow-left" className={styles.returnButton} />
```

Comment:
Since this button doesn't have any text, it's a good idea to add `aria-label` for accessibility.